### PR TITLE
DOC:special/signal: Add examples for `berp_zeros` and `band_stop_obj`

### PR DESCRIPTION
**Pull request by elleem**
#### Reference issue
Related : #7168

### CODE REVIEW

**Comment by j-bowhay** [comment]
Thanks @elleem! It would be great if you could add some context/motivation in these examples to aid understanding in addition to the execution of the code. Also, I would recommend only touching one module per pr as there isn't much overlap between signal and special maintainers, for example.

Finally, as these are documentation only changes we don't need to run the whole test suite so in future please could you add `[skip actions] [skip cirrus]` to the end of your commit as per https://docs.scipy.org/doc/scipy/dev/contributor/continuous_integration.html#skipping

**Comment by elleem** [comment]
Thank you so much for helping a beginner! I have tried to improve based on your feedback. :) Thanks again.

File: scipy/special/_basic.py [hunk 1]
```diff
@@ -2472,6 +2472,20 @@ def berp_zeros(nt):
            Functions", John Wiley and Sons, 1996.
            https://people.sc.fsu.edu/~jburkardt/f77_src/special_functions/special_functions.html
 
+    Example
+    -------
```
Comment (dschmitz89) [inline scipy/special/_basic.py:2476]:
```suggestion
    Examples
    --------
```

File: scipy/signal/_wavelets.py [hunk 2]
```diff
@@ -89,6 +89,30 @@ def qmf(hk):
     array_like
         High-pass filter coefficients.
 
+    Examples
+    --------
+    low_pass_coefficients = [0.48296, 0.8365, 0.22414, -0.12940]
+    high_pass_coefficients = qmf(low_pass_coefficients)
+    high_pass_coefficients
+    array([ 0.12940,  0.22414, -0.8365 ,  0.48296])
```
Comment (dschmitz89) [inline scipy/signal/_wavelets.py:97]:
```suggestion
    >>> from scipy.signal import qmf
    >>> low_pass_coefficients = [0.48296, 0.8365, 0.22414, -0.12940]
    >>> high_pass_coefficients = qmf(low_pass_coefficients)
    >>> high_pass_coefficients
    array([ 0.12940,  0.22414, -0.8365 ,  0.48296])
```

File: scipy/signal/_filter_design.py [hunk 3]
```diff
@@ -3654,6 +3654,27 @@ def band_stop_obj(wp, ind, passb, stopb, gpass, gstop, type):
...
         Filter order (possibly non-integer).
 
+    Example
+    -------
+    Band-stop filters are used in applications where certain frequency
+    components need to be blocked while others are allowed. For instance,
+    removing noise at specific frequencies while allowing the desired signal
+    to pass through. The order of a filter often determines its complexity and
+    accuracy. Determining the right order can be a challenge. This function
+    aims to estimate the order for an analog band stop filter:
```
Comment (dschmitz89) [inline scipy/signal/_filter_design.py:3664]: This should be shortened or become part of a `Notes` section.

File: scipy/signal/_filter_design.py [hunk 4]
```diff
@@ -3654,6 +3654,27 @@ def band_stop_obj(wp, ind, passb, stopb, gpass, gstop, type):
...
+
+    wp = 2
+    ind = 1
+    passb = np.array([1, 3])
+    stopb = np.array([0.5, 4])
+    gstop = 30
+    gpass = 3
+    filter_type = 'butter'
+    band_stop_obj(wp, ind, passb, stopb, gpass, gstop, filter_type)
+    print(result)  # Expected: 2.386294361119891
```
Comment (dschmitz89) [inline scipy/signal/_filter_design.py:3676]:
```suggestion
    import numpy as np
    from scipy.signal import band_stop_obj

    >>> wp = 2
    >>> ind = 1
    >>> passb = np.array([1, 3])
    >>> stopb = np.array([0.5, 4])
    >>> gstop = 30
    >>> gpass = 3
    >>> filter_type = 'butter'
    >>> band_stop_obj(wp, ind, passb, stopb, gpass, gstop, filter_type)
```

File: scipy/special/_basic.py [hunk 5]
```diff
@@ -2472,6 +2472,20 @@ def berp_zeros(nt):
...
+    crucial in various fields of science and engineering. For instance, if
+    you're working on problems related to heat conduction, wave propagation,
+    or certain types of differential equations, you may come across the Kelvin
+    function. Here's how you can compute the first 5 zeros of its derivative:
+
+    import numpy as np
+
+    Compute the first 5 zeros of the derivative of the Kelvin function ber
+    zeros = berp_zeros(5)
+    print("First 5 zeros of the derivative of ber:", zeros)
```
Comment (dschmitz89) [inline scipy/special/_basic.py:2487]:
```suggestion
    Compute the first 5 zeros of the derivative of the Kelvin function ber:

    >>> import numpy as np
    >>> from scipy.special import berp_zeros
    >>> berp_zeros(5)
    array([ 6.03871081, 10.51364251, 14.96844542, 19.41757493, 23.86430432])
```

File: scipy/special/_basic.py (hunk 1 above)
Comment (dschmitz89) [inline scipy/special/_basic.py:2476]:
```suggestion
    Examples
    --------
```

**Review by dschmitz89 — CHANGES_REQUESTED** [review]
Thanks for the contribution @elleem. Some advice from someone who originally started contributing to SciPy also mainly on documentation.

1. You probably noticed the errors in CI: to reproduce these locally, you need to run (for all possible options, check `python dev.py --help`):
`python dev.py refguide-check`

This will help you to correct things locally. Relying fully on CI for testing is not possible for SciPy as we would run out of the free quotas.

2. Your examples are missing the necessary imports from scipy (check suggestions).

3. For SciPy documentation, the order is typically first the "Notes" and then "Examples" section.

4. We typically do not use `print()` in the examples. The examples are meant to resemble an ipython section. There, the output of a function, is directly printed. See for example one example in `special` [here](https://docs.scipy.org/doc/scipy/reference/generated/scipy.special.jv.html#scipy.special.jv):
image
Code lines have to start with `>>> `, the printed outputs can directly be inserted.

**Comment by dschmitz89** [comment]
@elleem : friendly ping, are you still interested in finishing this PR?

**Comment by elleem** [comment]
Yes, I would really love to finish this PR. I got a new job and I have been underwater on my work. If possible, I would really appreciate a little extension and will prioritize getting this done.

**Comment by dschmitz89** [comment]
> Yes, I would really love to finish this PR. I got a new job and I have been underwater on my work. If possible, I would really appreciate a little extension and will prioritize getting this done.

Awesome, more time is of course possible. I would recommend to split the PR into one PR per documented function though. That will make review much easier.

**Comment by mdhaber** [comment]
@elleem Thanks for submitting this. Were you still interested in finishing? I can go ahead and make the requested changes so this makes it into the next version of SciPy, if you'd prefer.

File: scipy/signal/_wavelets.py [hunk 6]
```diff
@@ -89,6 +89,30 @@ def qmf(hk):
...
+    The QMF technique works by reversing the coefficients and alternating the
+    signs to create the high-pass filter. This is particularly useful when
+    designing filter banks where the low-pass and high-pass filters need to
+    have certain relationships for proper signal decomposition and
+    reconstruction.
+
+    References
+    ----------
+    1. Vaidyanathan, P.P. (1993). "Multirate Systems and Filter Banks".
+       Prentice Hall. ISBN: 978-0136057185.
```
Comment (mdhaber) [inline scipy/signal/_wavelets.py:115]:
```suggestion
    .. [1] Vaidyanathan, P.P. (1993). "Multirate Systems and Filter Banks".
           Prentice Hall. ISBN: 978-0136057185.
```

File: scipy/signal/_filter_design.py (hunk 3 above)
Comment (mdhaber) [inline scipy/signal/_filter_design.py:3664]:
```suggestion
    Notes
    -----
    Band-stop filters are used in applications where certain frequency
    components need to be blocked while others are allowed. For instance,
    removing noise at specific frequencies while allowing the desired signal
    to pass through. The order of a filter often determines its complexity and
    accuracy. Determining the right order can be a challenge. This function
    aims to estimate the order for an analog band stop filter.
    
    Examples
    --------
```

File: scipy/signal/_wavelets.py [hunk 7]
```diff
@@ -89,6 +89,31 @@ def qmf(hk):
...
+    The QMF technique works by reversing the coefficients and alternating the
+    signs to create the high-pass filter. This is particularly useful when
+    designing filter banks where the low-pass and high-pass filters need to
+    have certain relationships for proper signal decomposition and
+    reconstruction.
+
+    References
+    ----------
+    .. [1] Vaidyanathan, P.P. (1993). "Multirate Systems and Filter Banks".
+           Prentice Hall. ISBN: 978-0136057185.
```
Comment (mdhaber) [inline scipy/signal/_wavelets.py:116]:
```suggestion
    Notes
    -----
    This function takes the coefficients of a low-pass filter `hk` and computes
    the corresponding high-pass filter coefficients using a quadrature mirror
    filter (QMF) technique. The QMF technique is commonly used in filter bank
    design, especially in wavelet transform applications.

    The QMF technique works by reversing the coefficients and alternating the
    signs to create the high-pass filter. This is particularly useful when
    designing filter banks where the low-pass and high-pass filters need to
    have certain relationships for proper signal decomposition and
    reconstruction.

    References
    ----------
    .. [1] Vaidyanathan, P.P. (1993). "Multirate Systems and Filter Banks".
           Prentice Hall. ISBN: 978-0136057185.
           
    Examples
    --------
    >>> from scipy.signal import qmf
    >>> low_pass_coefficients = [0.48296, 0.8365, 0.22414, -0.12940]
    >>> high_pass_coefficients = qmf(low_pass_coefficients)
    >>> high_pass_coefficients
    array([ 0.12940,  0.22414, -0.8365 ,  0.48296])

```

File: scipy/special/_basic.py [hunk 8]
```diff
@@ -2472,6 +2472,21 @@ def berp_zeros(nt):
...
 
+    Examples
+    --------
+    Understanding the behavior of special functions and their derivatives is
+    crucial in various fields of science and engineering. For instance, if
+    you're working on problems related to heat conduction, wave propagation,
+    or certain types of differential equations, you may come across the Kelvin
+    function. Here's how you can compute the first 5 zeros of its derivative:
+
+    Compute the first 5 zeros of the derivative of the Kelvin function ber:
```
Comment (mdhaber) [inline scipy/special/_basic.py:2483]:
```suggestion
    Notes
    -----
    Understanding the behavior of special functions and their derivatives is
    crucial in various fields of science and engineering. For instance, if
    you're working on problems related to heat conduction, wave propagation,
    or certain types of differential equations, you may come across the Kelvin
    function.

    Examples
    --------
    Compute the first 5 zeros of the derivative of the Kelvin function
    `scipy.special.ber`:
```

File: scipy/special/_basic.py [hunk 9]
```diff
@@ -2494,6 +2494,22 @@ def berp_zeros(nt):
            Functions", John Wiley and Sons, 1996.
            https://people.sc.fsu.edu/~jburkardt/f77_src/special_functions/special_functions.html
 
+    Notes
+    -----
+    Understanding the behavior of special functions and their derivatives is
+    crucial in various fields of science and engineering. For instance, if
+    you're working on problems related to heat conduction, wave propagation,
+    or certain types of differential equations, you may come across the Kelvin
+    function.
```
Comment (j-bowhay) [inline scipy/special/_basic.py:2503]: Thanks for pushing this on, @mdhaber. I have to admit I find this paragraph slightly too generic. Would you object to removing it?

Comment (j-bowhay) [reply scipy/special/_basic.py:2503]:
```suggestion
```

**Review by j-bowhay — APPROVED** [review]

**Pull request merged**